src/modules/esl-utils/dom/scroll/test/utils.test.ts

```diff
@@ -0,0 +1,258 @@
+  jest.spyOn(el, 'clientWidth', 'get').mockImplementation(() => 99);
+};
+
+describe('Function hasVerticalScroll', () => {
+  test('should be vertical scroll on default element', () => {
+    mockElHeight($html);
+    expect(hasVerticalScroll()).toBeTruthy();
+  });
+
+  test('should be vertical scroll on target element', () => {
+    const target = document.createElement('div');
+    mockElHeight(target);
+    expect(hasVerticalScroll(target)).toBeTruthy();
+  });
+});
+
+describe('Function hasHorizontalScroll', () => {
+  test('should be horizontal scroll on default element', () => {
+    mockElWidth($html);
+    expect(hasHorizontalScroll()).toBeTruthy();
+  });
+
+  test('should be horizontal scroll on target element', () => {
+    const target = document.createElement('div');
+    mockElWidth(target);
+    expect(hasHorizontalScroll(target)).toBeTruthy();
+  });
+});
+
+describe('Function isScrollLocked', () => {
+  const target = document.createElement('div');
+  target.style.overflow = 'auto';
+  describe('Locked Scroll', () => {
+    beforeAll(() => lockScroll(target));
+    test('lock attribute should be set', () => expect(target.hasAttribute('esl-scroll-lock')).toBeTruthy());
+    test('target should be locked', () => expect(isScrollLocked(target)).toBeTruthy());
+  });
+
+  describe('Unlocked Scroll', () => {
+    beforeAll(() => unlockScroll(target));
+    test('lock attribute should be removed', () => expect(target.hasAttribute('esl-scroll-lock')).toBeFalsy());
+    test('target should be locked', () => expect(isScrollLocked(target)).toBeFalsy());
+  });
+});
+
+describe('Function lockScroll', () => {
+  const target = document.createElement('div');
+  $body.style.overflow = 'auto';
+  $html.style.overflow = 'auto';
+
+  test('target`s parent should be locked', () => {
+    lockScroll(target);
+    expect(isScrollLocked($body)).toBeTruthy();
+  });
+
+  test('target should be locked', () => {
+    target.style.overflow = 'auto';
+    lockScroll(target);
+    expect(isScrollLocked(target)).toBeTruthy();
+  });
+
+  test('default element should be locked', () => {
+    lockScroll();
+    expect(isScrollLocked($html)).toBeTruthy();
+  });
+
+  describe('Scroll lock with recursive option', () => {
+    unlockScroll(target);
+    unlockScroll($body);
+    jest.spyOn(target, 'parentElement', 'get').mockImplementation(() => $body);
+
+    lockScroll(target, {recursive: true});
+    test('target should be locked', () => expect(isScrollLocked(target)).toBeTruthy());
+    test('target`s parent should be locked', () => expect(isScrollLocked($body)).toBeTruthy());
+  });
+
+  describe('Lock with initiator', () => {
+    describe('Lock with initiator should be successful', () => {
+      beforeAll(() => {
+        jest.spyOn(target, 'setAttribute');
+        lockScroll(target, {initiator: 'init'});
+      });
+
+      test('setAttribute should have been called', () => expect(target.setAttribute).toHaveBeenCalled());
```

Comment:
Cosmetic: we can check that in bounds of the isScrollLocked unit, in case it handles attribute, then we can rely on it in the future tests. So basically, from this thesis perspective, we are interested in the "locking of the scroll" but not in the details of how it is locked.